### DISCUSSION TITLE
Error   | Mesh contains a singular tetrahedron with radius    0.0000000000000000

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#937] [Q&A] [answered]
For the same mesh（gmsh2.2）, when I use pumgen to convert the format of the mesh after scaling it up by a factor of 1000, I find that `Minimum insphere found: 0`. When simulating the mesh, I get `Error | Mesh contains a singular tetrahedron with radius    0.0000000000000000` . Is this due to pumgen's limitation on the size range of values?

![image](https://github.com/SeisSol/SeisSol/assets/114474894/ca53eb5d-13cc-42dc-867f-b787190d8650)
![image](https://github.com/SeisSol/SeisSol/assets/114474894/b38e3d01-d9c5-49fa-bf0d-1ed0f649963e)

**Comment by sebwolf-de** [comment]
Your mesh has around 7 million elements. PUMGen definitely works in that size range! I suspect that your specific mesh indeed has a degraded element in it.

**Comment by buzhiqiuyuebai** [comment]
> Your mesh has around 7 million elements. PUMGen definitely works in that size range! I suspect that your specific mesh indeed has a degraded element in it.

I mean maybe the mesh is too big, for example mesh size is 1e8, would that affect the results? Because I eliminated the problem by simply shrinking this mesh a thousand times.

**Reply by sebwolf-de** [reply]
We have used PUMGen for meshes up to 500 million elements, so I don't think this is a problem inherent to PUMGen.
What exactly do you mean by too big? The number of elements or the domain the mesh covers?
What exactly do you mean by shrink? The number of elements or the size of the domain (e.g. kilometers -> meters)?

**Comment by buzhiqiuyuebai** [comment]
> We have used PUMGen for meshes up to 500 million elements, so I don't think this is a problem inherent to PUMGen. What exactly do you mean by too big? The number of elements or the domain the mesh covers? What exactly do you mean by shrink? The number of elements or the size of the domain (e.g. kilometers -> meters)?

I mean  the size of the domain (e.g. kilometers -> meters)

**Reply by sebwolf-de** [reply]
Without the raw meshes, I can't say much. Maybe some over-/underflow here: https://github.com/SeisSol/PUMGen/blob/3fb1875df66c7ce14790af7d6dd3521a6ef71fcb/src/pumgen.cpp#L202

**Reply by buzhiqiuyuebai** [reply, marked as answer]
Maybe poor mesh quality leads to this result.
